src/puppetlabs/trapperkeeper/authorization/acl.clj

```diff
@@ -79,30 +76,73 @@
   [pattern]
   (-> pattern (str/lower-case) (str/split #"\.") reverse vec))
 
-(def allow-all { :auth-type :allow :type :allow-all :qualifier :exact :length nil :pattern [] })
+(def allow-all {:auth-type :allow
+                :type :allow-all
+                :qualifier :exact
+                :length nil
+                :pattern []})
+
+(def deny-all {:auth-type :deny
+               :type :regex
+               :qualifier :exact
+               :length nil
+               :pattern #"^/.*/$"})
 
 (schema/defn new-domain :- Entry
   "Creates a new ACE for a domain"
   [type :- auth-type
    pattern :- schema/Str]
   (cond
-    ; global allow
-    (= "*" pattern) allow-all
+    (and (= "*" pattern) (= type :allow))
+    allow-all
+
+    ; global deny
+    (and (= "*" pattern) (= type :deny))
+    deny-all
```

Comment:
Yeah, this was my thinking exactly. I didn't want to spend a whole bucket of time on straightening out the logic in this PR, since we're going to revisit it later. But inlining wouldn't be a bad idea.